From 023bb6d0407f67cd4e474355f14fe2ff9b0f050d Mon Sep 17 00:00:00 2001
From: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Date: Wed, 6 Mar 2019 19:04:06 +0200
Subject: [PATCH] u-boot-qoriq: Fix misuse of 'IS_ENABLED' macro

In the environment driver of u-boot-qoriq 2017.12 in order to get
the location of the environment, specific CONFIG_ENV_IS_IN_* have been
defined. However, they are incorrectly checked using 'IS_ENABLED(value)' macro
which expects parameter 'value' to be '1' or '0'.
Since those CONFIG_ENV_IS_IN_* take values of 'y', 'm' or 'n', the preprocessor
always evaluates 'IS_ENABLED(CONFIG_ENV_IS_IN_*)' to '0', causing the 'env_init'
function to not be correctly executed and u-boot to hang.
The proposed solution is to use '#ifdef CONFIG_ENV_IS_IN_*' instead of
'IS_ENABLED' macro, in accordance with version 2018.03 of u-boot-qoriq.

Issue: ALB-3494
Signed-off-by: Ciprian Costea <ciprianmarian.costea@nxp.com>
---
 env/env.c | 31 ++++++++++++++++++++-----------
 1 file changed, 20 insertions(+), 11 deletions(-)

diff --git a/env/env.c b/env/env.c
index 76a5608..d72e0df 100644
--- a/env/env.c
+++ b/env/env.c
@@ -28,27 +28,36 @@ static struct env_driver *env_driver_lookup(enum env_location loc)
 
 static enum env_location env_get_default_location(void)
 {
-	if IS_ENABLED(CONFIG_ENV_IS_IN_EEPROM)
+#ifdef CONFIG_ENV_IS_IN_EEPROM
 		return ENVL_EEPROM;
-	else if IS_ENABLED(CONFIG_ENV_IS_IN_FAT)
+#endif
+#ifdef CONFIG_ENV_IS_IN_FAT
 		return ENVL_FAT;
-	else if IS_ENABLED(CONFIG_ENV_IS_IN_FLASH)
+#endif
+#ifdef CONFIG_ENV_IS_IN_FLASH
 		return ENVL_FLASH;
-	else if IS_ENABLED(CONFIG_ENV_IS_IN_MMC)
+#endif
+#ifdef CONFIG_ENV_IS_IN_MMC
 		return ENVL_MMC;
-	else if IS_ENABLED(CONFIG_ENV_IS_IN_NAND)
+#endif
+#ifdef CONFIG_ENV_IS_IN_NAND
 		return ENVL_NAND;
-	else if IS_ENABLED(CONFIG_ENV_IS_IN_NVRAM)
+#endif
+#ifdef CONFIG_ENV_IS_IN_NVRAM
 		return ENVL_NVRAM;
-	else if IS_ENABLED(CONFIG_ENV_IS_IN_REMOTE)
+#endif
+#ifdef CONFIG_ENV_IS_IN_REMOTE
 		return ENVL_REMOTE;
-	else if IS_ENABLED(CONFIG_ENV_IS_IN_SPI_FLASH)
+#endif
+#ifdef CONFIG_ENV_IS_IN_SPI_FLASH
 		return ENVL_SPI_FLASH;
-	else if IS_ENABLED(CONFIG_ENV_IS_IN_UBI)
+#endif
+#ifdef CONFIG_ENV_IS_IN_UBI
 		return ENVL_UBI;
-	else if IS_ENABLED(CONFIG_ENV_IS_NOWHERE)
+#endif
+#ifdef CONFIG_ENV_IS_NOWHERE
 		return ENVL_NOWHERE;
-	else
+#endif
 		return ENVL_UNKNOWN;
 }
 
-- 
1.9.1

